packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.ts

```diff
@@ -243,7 +243,11 @@ export async function suggest(
     if (!ast.length) {
       return suggestions.filter(isSourceCommand);
     }
-    return suggestions.filter((def) => !isSourceCommand(def));
+
+    // Temporarily disabling INLINESTATS from being suggested until it's released
+    // TODO: Remove in filter when command is available in tech preview
+    // See  https://github.com/elastic/kibana/issues/189356
+    return suggestions.filter((def) => !isSourceCommand(def) && def.label !== 'INLINESTATS');
```

Comment:
Should we add a `hidden` property to the command definition interface? It seems like it might be worth building a feature-flag pattern for commands since Elasticsearch seems to follow the pattern of adding commands on snapshots first so we have a period of time where we don't show them.

We'd have to be careful, but we could even consider detecting whether we're on a snapshot version and conditionally showing these.

cc @stratoula 

For the purposes of this PR though, you could at least consider the `hidden` command property idea. Not a requirement though.

Reply:
Yes Drew this is a very good idea. Moving things to snapshot and then to tech preview seems to be a process that ES is going to follow for some time. 

On our side we want to support validation on these snapshot features but not autocomplete. So it would be cool our architecture to support it. 

@qn895 can you add this in the PR?